udata/core/topic/models.py

```diff
@@ -23,10 +26,30 @@ class Topic(db.Document):
     owner = db.ReferenceField('User')
     featured = db.BooleanField()
     private = db.BooleanField()
+    extras = db.ExtrasField()
 
     def __str__(self):
         return self.name
 
+    @classmethod
+    def pre_save(cls, sender, document, **kwargs):
+        try:
+            original_doc = sender.objects.get(id=document.id)
+            datasets_list_dif = list(set(original_doc.datasets).symmetric_difference(set(document.datasets)))
+            reuses_list_dif = list(set(original_doc.reuses).symmetric_difference(set(document.reuses)))
+        except cls.DoesNotExist:
+            datasets_list_dif = document.datasets
+            reuses_list_dif = document.reuses
+        if datasets_list_dif:
+            for dataset in datasets_list_dif:
+                reindex.delay(*as_task_param(dataset))
+        if reuses_list_dif:
+            for reuse in reuses_list_dif:
+                reindex.delay(*as_task_param(reuse))
```

Comment:
We actually haven't discussed indexing reuses with topic info. It would makes sense since we do for datasets, but we also have a `topic` property in the Reuse object that might get in conflict.